src/sketch.js

```diff
@@ -5,54 +5,108 @@
 // sound controller container
 let soundOn = false;
 
-// perlin noise terrain container
-let terrain;
-
 // soundtrack p5.SoundFile container
 let soundtrack;
 
+// webcam stream container
+let webcam;
+
+// webcam controller container
+let webcamOn = false;
+
+// perlin noise terrain container
+let terrain;
+
 /**
  * I draw this sketch sound indicator.
- * 
+ *
+ * Args:
+ *  indicator (object): indicator image
+ *  toggled (boolean): indicator toggle status
+ *  position (number): position in the indicators stack (1 is lowest)
+ *
  * Returns:
  *  undefined.
  */
-function drawSoundIndicator()
+function drawIndicator(indicator, toggled, position)
 {
     // push transformation matrix
     push();
 
-    // translate to bottom right corner
-    translate((WIDTH / 2) - 50, (HEIGHT / 2) - 50);
+    // translate to indicators stack
+    translate((WIDTH / 2) - 50, (HEIGHT / 2) - position * 50);
 
     // set indicator transparency
-    soundOn ? tint(255, 100) : tint(255, 255);
-   
+    toggled ? tint(255, 255) : tint(255, 100);
+
     // draw image
-    image(keyboard, 0, 0, 50, 50);
-    
+    image(indicator, 0, 0, 50, 50);
+
     // pop transformation matrix
     pop();
 }
 
 /**
+ * Draw webcam input on sketch.
+ *
+ * Args:
+ *  position (number): position on indicators stack
+ *
+ * Returns:
+ *  undefined.
+ */
+function drawWebcamInput(position)
+{
+    // webcam is activated: draw input
+    if(webcamOn)
+    {
+        // translate to indicators stack
+        translate((WIDTH / 2) - 50, (HEIGHT / 2) - position * 50);
+
+        // apply transparency
+        tint(200, 100);
+
+        // draw webcam input
+        image(webcam, 0, 0, 50, 50);
+    }
+}
+
+/**
+ * I update the webcam functioning status.
+ *
+ * Returns:
+ *  undefined.
+ */
+function updateWebcamStatus()
+{
+    webcamOn = webcam.loadedmetadata;
+}
```

Comment:
This is not a boolean anymore. Please extract as separate variable to contain webcamImage.